config/config.js

```diff
@@ -68,10 +69,20 @@ var validateEnvironmentVariable = function () {
   console.log(chalk.white(''));
 };
 
-/** Validate config.domain is set
+/**
+ * Validate user is running supported Node.js version
+ */
+var validateNodeJsVersion = function (config) {
+  if (!semver.satisfies(process.version, config.meanjs.engines.node)) {
+    console.log(chalk.red('+ Important warning: your version of Node (' + process.version.substr(1) + ') is unsupported by MEAN.js. Please use a version that satisfies the following requirements: ' + config.meanjs.engines.node + ', as specified in the package.json file.'));
+  }
+};
+
+/**
+ * Validate config.domain is set
  */
 var validateDomainIsSet = function (config) {
-  if (!config.domain) {
+  if (!config.domain && process.env.NODE_ENV === 'production') {
```

Comment:
I see what you're saying @hyperreality. However, maybe we could just "introduce" them to the idea of the domain setting needing to be set; can we show this message as a warning in non-Production environments, and display as a critical (red) error when in production?

I don't want to over-complicate that matter, but I do think we can address the underlying issue of user's getting confused. My statement of "scary" was more that after some time spent on development, with everything working as expected, and to all of a sudden see a message like that in a production environment (whether local or deployed) could give someone a moment of panic. Either way, we can pick this up in a separate PR.